Packages/com.unity.inputsystem/InputSystem/Plugins/EnhancedTouch/TouchSimulation.cs

```diff
@@ -0,0 +1,437 @@
+using System;
+using UnityEngine.InputSystem.Controls;
+using UnityEngine.InputSystem.LowLevel;
+using UnityEngine.InputSystem.Utilities;
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEngine.InputSystem.Editor;
+#endif
+
+////TODO: add pressure support
+
+////REVIEW: extend this beyond simulating from Pointers only? theoretically, we could simulate from any means of generating positions and presses
+
+////REVIEW: I think this is a workable first attempt but overall, not a sufficient take on input simulation. ATM this uses InputState.Change
+////        to shove input directly into Touchscreen. Also, it uses state change notifications to set off the simulation. The latter leads
+////        to touch input potentially changing multiple times in response to a single pointer event. And the former leads to the simulated
+////        touch input not being visible at the event level -- which leaves Touch and Finger slightly unhappy, for example.
+////        I think being able to cycle simulated input fully through the event loop would result in a setup that is both simpler and more robust.
+////        Also, it would allow *disabling* the source devices as long as we don't disable them in the backend, too.
+////        Finally, the fact that we spin off input *from* events here and feed that into InputState.Change() by passing the event along
+////        means that places that make sure we process input only once (e.g. binding composites which will remember the event ID they have
+////        been triggered from) may reject the simulated input when they have already seen the non-simulated input (which may be okay
+////        behavior).
+
+namespace UnityEngine.InputSystem.Touch
+{
+    /// <summary>
+    /// Adds a <see cref="Touchscreen"/> with input simulated from other types of <see cref="Pointer"/> devices (e.g. <see cref="Mouse"/>
+    /// or <see cref="Pen"/>).
+    /// </summary>
+    [AddComponentMenu("Input/Touch Simulation")]
+    [ExecuteInEditMode]
+    #if UNITY_EDITOR
+    [InitializeOnLoad]
+    #endif
+    public class TouchSimulation : MonoBehaviour, IInputStateChangeMonitor
+    {
+        public Touchscreen simulatedTouchscreen { get; private set; }
+
+        public static TouchSimulation instance => s_Instance;
+
+        public static void Enable()
+        {
+            if (instance == null)
+            {
+                ////TODO: find instance
```

Comment:
Guess this TODO should not be lost, I suppose this might break apart when you have both a MonoBehaviour you added to the scene _and_ call the static `Enable` function somewhere. But same question as for EnhancedTouch - do we need both methods to enable this?